### ISSUE TITLE
[Bug]: Can't create new API key and shown key is of other user

### ISSUE THREAD

**Issue by panzi** [infiniflow/ragflow#4138]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### RAGFlow workspace code commit ID

834c4d81f38d18704684f0d74ec235368b080023

### RAGFlow image version

1b817a5b(v0.14.1~62) slim

### Other environment information

```Markdown
OS type: Linux
```

### Actual behavior

The "Create new key" button is grayed out/disabled for my user for some reason. There is one API key entry already there, but that is by another user. If I use that anything created via the API belongs to that other user. When I click delete, it acts as if it would delete that API key, but it actually isn't deleted. I don't think I should be able to see things of other users! Also if I know the ID of a dataset created that way I can just access it without the other user sharing it with me.

![Bildschirmfoto_20241219_230055](https://github.com/user-attachments/assets/c52be2ef-5f0d-4971-8a2c-2a66a852a60b)

### Expected behavior

- I should only be able to see my API keys
- I should be able to create an API key
- I shouldn't be able to access the dataset of another user via a known URL

### Steps to reproduce

```Markdown
Not sure if this is actually enough, but what we did is:

- Created user Alice
- Alice created datasets, added OpenAI API keys, generated a Ragflow API key
- Created user Bob
- Alice shared *some* datasets with Bob
- Bob used the existing API key to create more datasets, documents, and chat assistants
- Bob can't see those things in his web interface and can't create another API key
```

**Comment by KevinHuSh** [comment]
One tenant could only use one API key, like OpenAI or some others.
As lone as it's been re-created, all the API/SDK invokation will be unauthorized.

**Comment by panzi** [comment]
What is a tenant? The API key that Bob sees was created by Alice and will create things that belong to Alice. Bob can't create another API key for themselves.

>  As lone as it's been re-created, all the API/SDK invokation will be unauthorized.

I don't understand this sentence. Can you rephrase that? As long what is re-created? What is unauthorized exactly?

**Comment by KevinHuSh** [comment]
Once you delete the API key, all the applications using that API key will be unauthorized to invoke.

**Comment by KevinHuSh** [comment]
Bob needs to register another account which generates its own API key.

**Comment by panzi** [comment]
> Bob needs to register another account which generates its own API key.

And that's where the bug is. I did register another account, but can't generate my own API key. I only see the API key of the other user.

> Once you delete the API key, all the applications using that API key will be unauthorized to invoke.

Of course. But the delete button does nothing. Because it's not my API key, even though it is displayed for me! And the button to create another key is disabled.
